samples/bluetooth/peripheral_ams_client/src/main.c

```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/gatt_dm.h>
+#include <bluetooth/services/ams_client.h>
+
+#include <settings/settings.h>
+
+#include <dk_buttons_and_leds.h>
+
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN (sizeof(DEVICE_NAME) - 1)
+
+#define RUN_STATUS_LED DK_LED1
+#define CON_STATUS_LED DK_LED2
+#define RUN_LED_BLINK_INTERVAL 1000
+
+#define MAX_MSG_SIZE 100
+
+enum {
+	IS_UPDATE_TRACK,
+	HAS_NEXT_TRACK,
+	HAS_PREVIOUS_TRACK
+};
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_LIMITED | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_SOLICIT128, BT_UUID_AMS_VAL),
+};
+
+static const struct bt_data sd[] = {
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+};
+
+static struct bt_ams_client ams_c;
+
+static const enum bt_ams_player_attribute_id entity_update_player[] = {
+	BT_AMS_PLAYER_ATTRIBUTE_ID_NAME,
+	BT_AMS_PLAYER_ATTRIBUTE_ID_PLAYBACK_INFO,
+	BT_AMS_PLAYER_ATTRIBUTE_ID_VOLUME
+};
+
+static const enum bt_ams_track_attribute_id entity_update_track[] = {
+	BT_AMS_TRACK_ATTRIBUTE_ID_ARTIST,
+	BT_AMS_TRACK_ATTRIBUTE_ID_ALBUM,
+	BT_AMS_TRACK_ATTRIBUTE_ID_TITLE,
+	BT_AMS_TRACK_ATTRIBUTE_ID_DURATION
+};
+
+static atomic_t appl_flags;
+
+static char msg_buff[MAX_MSG_SIZE + 1];
+
+static void ea_read_cb(struct bt_ams_client *ams_c, uint8_t err, const uint8_t *data, size_t len)
+{
+	if (err) {
+		printk("AMS EA read error 0x%02X\n", err);
+	} else if (len > MAX_MSG_SIZE) {
+		printk("AMS EA data size is too big\n");
+	} else {
+		memcpy(msg_buff, data, len);
+		msg_buff[len] = '\0';
+		printk("AMS EA: %s\n", msg_buff);
+	}
+}
+
+static void ea_write_cb(struct bt_ams_client *ams_c, uint8_t err)
+{
+	if (!err) {
+		err = bt_ams_read_entity_attribute(ams_c, ea_read_cb);
+	} else {
+		printk("AMS EA write error 0x%02X\n", err);
+	}
+}
+
+static void notify_rc_cb(struct bt_ams_client *ams_c,
+			 const uint8_t *data, size_t len)
+{
+	char str_hex[4];
+	bool has_next_track = false;
+	bool has_previous_track = false;
+	enum bt_ams_remote_command_id cmd_id;
+
+	if (len > 0) {
+		if (len * 3 > MAX_MSG_SIZE) {
```

Comment:
Could you explain why you multiply here in comment ?

Reply:
Will do.

The check will be more accurate modifying to:

    if (len * 3 - 1 > MAX_MSG_SIZE) {

The first data byte conversion does not add a comma.